### PR TITLE
[Doc][KubeRay] Upgrade RayService with GCS fault tolerance enabled without downtime

### PR DESCRIPTION
**Pull request by kevin85421**
## Why are these changes needed?

https://github.com/ray-project/kuberay/issues/1296

## Related issue number

https://github.com/ray-project/kuberay/issues/1296

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by architkulkarni — APPROVED** [review]

File: doc/source/cluster/kubernetes/troubleshooting/rayservice-troubleshooting.md [hunk 1]
```diff
@@ -334,4 +335,22 @@ This will trigger a new RayCluster preparation instead of an in-place update.
...
-You can refer to [kuberay#1297](https://github.com/ray-project/kuberay/issues/1297) for more details.
\ No newline at end of file
+You can refer to [kuberay#1297](https://github.com/ray-project/kuberay/issues/1297) for more details.
+
+(kuberay-raysvc-issue10)=
+### Issue 10: Upgrade RayService with GCS fault tolerance enabled without downtime
+
+KubeRay employs the value of the annotation [ray.io/external-storage-namespace](kuberay-external-storage-namespace) to assign the environment variable `RAY_external_storage_namespace` to all Ray Pods managed by the RayCluster.
+This value represents the storage namespace in Redis where the Ray cluster metadata resides.
+In the process of a head Pod recovery, it attempts to reconnect to the Redis server using the `RAY_external_storage_namespace` value to recover the cluster data.
```
[the remark this answers is not in the document]
Comment (angelinalg) [reply doc/source/cluster/kubernetes/troubleshooting/rayservice-troubleshooting.md:345]: If this is incorrect, please correct with whatever `it` should be.

**Review by angelinalg — APPROVED** [review]
Just some nits.

**Comment by kevin85421** [comment]
Thanks, updated!

**Comment by architkulkarni** [comment]
test_object_assign_owner_client_mode test failure unrelated

**Pull request merged**